postprocess/extra-env.jsonnet

```
@@ -0,0 +1,55 @@
+/**
+ * Adjust StatefuleSet generated by helm template
+ */
+local com = import 'lib/commodore.libjsonnet';
+local inv = com.inventory();
+local params = inv.parameters.keycloak;
+
+local statefulset_file = std.extVar('output_path') + '/statefulset.yaml';
+local statefulset = com.yaml_load(statefulset_file);
+
+local insertMap(sup, map) = std.set(
+  [
+    {
+      name: k,
+    } + com.makeMergeable(map[k])
+    for k in std.objectFields(map)
+  ]
+  +
+  sup, function(c) c.name
+);
+
+{
+  statefulset: statefulset {
+    spec+: {
+      template+: {
+        spec+: {
+          containers: [
+            if c.name == 'keycloak' then
+              c {
+                env: insertMap(
+                  if 'env' in super then super.env else [],
```

If someone has a better solution to check if the field exists in `super`, I'm all ears.

---

Generally, `std.objectHas` should work with `super`

```suggestion
                  if std.objectHas(super, 'env') then super.env else [],
```

Or you could use the commodore.libjsonnet wrapper:

```suggestion
                  com.getValueOrDefault(super, 'env', []),
```

---

Sadly this fails for me with 
`RuntimeError: STATIC ERROR: /keycloak/postprocess/extra-env.jsonnet:31:36-41: expected . or [ after super.`

That's why i switched to using `in`

---

Theoretically `super` and `c` should be equivalent in this context, cf. https://jsonnet.org/ref/language.html#inheritance

---

Are you fine with leaving this as is? As far as I can tell the current solution is correct and the other option of using `com.getValueOrDefault` is IMO not more readable, especially for the other cases where I cannot simply use `c`

---

I'm fine to leave it as it is